capsules/extra/src/bmi270.rs

```diff
@@ -0,0 +1,523 @@
+                    State::Sleep => {
+                        buffer[0] = Registers::PwrConf as u8;
+                        buffer[1] = 0x00_u8;
+
+                        if let Err((_error, buffer)) = self.i2c.write(buffer, 2) {
+                            self.buffer.replace(buffer);
+                            self.i2c.disable();
+                        } else {
+                            self.state.set(State::WaitingForAlarm(ALARM_TIME_450));
+                        }
+                    }
+                    State::Idle => {
+                        buffer[0] = Registers::Data8 as u8;
+
+                        if let Err((i2c_err, buffer)) = self.i2c.write(buffer, 1) {
+                            self.state.set(State::Sleep);
+                            self.buffer.replace(buffer);
+                            self.ninedof_client
+                                .map(|client| client.callback(i2c_err as usize, 0, 0));
+                        } else {
+                            self.state.set(State::Read);
+                        }
+                    }
+                    _ => {}
+                }
+            })
+            .ok_or(ErrorCode::FAIL)
+    }
+
+    fn handle_alarm(&self) {
+        let _ = self.buffer.take().map(|buffer| match self.state.get() {
+            State::WaitingForAlarm(us) => {
+                if us == ALARM_TIME_450 {
+                    buffer[0] = Registers::InitCtrl as u8;
+                    buffer[1] = 0x00_u8;
+
+                    if let Err((i2c_err, buffer)) = self.i2c.write(buffer, 2) {
+                        self.state.set(State::Sleep);
+                        self.buffer.replace(buffer);
+                        self.ninedof_client
+                            .map(|client| client.callback(i2c_err as usize, 0, 0));
+                    } else {
+                        self.state.set(State::InitWriteConfig);
+                    }
+                } else if us == ALARM_TIME_20000 {
+                    buffer[0] = Registers::InternalStatus as u8;
+
+                    if let Err((i2c_err, buffer)) = self.i2c.write_read(buffer, 1, 1) {
+                        self.state.set(State::Sleep);
+                        self.buffer.replace(buffer);
+                        self.ninedof_client
+                            .map(|client| client.callback(i2c_err as usize, 0, 0));
+                    } else {
+                        self.state.set(State::CheckStatus);
+                    }
+                }
+            }
+            _ => {}
+        });
+    }
+}
+
+impl<'a, A: Alarm<'a>, I: I2CDevice> NineDof<'a> for BMI270<'a, A, I> {
+    fn set_client(&self, client: &'a dyn NineDofClient) {
+        self.ninedof_client.set(client);
+    }
+
+    fn read_accelerometer(&self) -> Result<(), ErrorCode> {
+        if !self.pending_accel.get() {
+            self.pending_accel.set(true);
+            if self.pending_gyro.get() {
+                Err(ErrorCode::BUSY)
```

Comment:
I think this should return `Ok(())`. The caller doesn't know that there is a shared device doing both accel and gyro, and since there wasn't a previous pending request for accel this should return ok.